code/modules/reagents/chemistry/reagents.dm

```diff
@@ -41,6 +41,8 @@ GLOBAL_LIST_INIT(name2reagent, build_name2reagent())
 	var/reagent_weight = 1 //affects how far it travels when sprayed
 	var/metabolizing = FALSE
 	var/harmful = FALSE //is it bad for you? Currently only used for borghypo. C2s and Toxins have it TRUE by default.
+	//Are we from a material? We might wanna know that for special stuff. Like metalgen
+	var/material = null
```

Comment:
```suggestion
	var/material
```

Also you should be replacing the type in this var with a reference to the actual material instance